lib/buffer.js

```diff
@@ -379,6 +378,16 @@ Buffer.prototype.toString = function(encoding, start, end) {
 };
 
 
+Buffer.prototype.toString = function() {
+  const length = this.length | 0;
+  if (length === 0)
+    return '';
+  if (arguments.length === 0)
+    return this.utf8Slice(0, length);
+  return slowToString(this, arguments[0], arguments[1], arguments[2]);
```

Comment:
Okay, seems 4.2 does a pretty good job too, once it warms up.  I'll switch this over to `.apply()`.